sdk/state/update.go

```diff
@@ -97,9 +87,9 @@ func (c *Channel) PaymentTxs(p Payment) (close, decl *txnbuild.Transaction, err
 		ResponderEscrow:            c.responderEscrowAccount().Address,
 		StartSequence:              c.startingSequence,
 		IterationNumber:            c.NextIterationNumber(),
-		AmountToInitiator:          maxInt64(0, newBalance.Amount*-1),
-		AmountToResponder:          maxInt64(0, newBalance.Amount),
-		Asset:                      p.Amount.Asset,
+		AmountToInitiator:          maxInt64(0, ca.Balance.Amount*-1),
```

Comment:
FYI you can also do the following (e.g. https://play.golang.org/p/Qo8v2N_PVN7):
```suggestion
		AmountToInitiator:          maxInt64(0, -ca.Balance.Amount),
```

Reply:
til! I actually think it reads easier with the `*-1` though? Unless in golang that's normal. Going to leave for now can change later if needed